lib/transaction.js

```diff
@@ -46,6 +47,7 @@ Transaction.hookTypes = {
  * @returns {*}
  */
 Transaction.prototype.commit = function(cb) {
+  cb = cb || createPromiseCallback();
   return this.connector.commit(this.connection, cb);
```

Comment:
I think you need to return `cb.promise`.

Reply:
See an example at https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L319

Reply:
Thanks @raymondfeng 🙇 that is correct.